### DISCUSSION TITLE
Invalid hook call when creating npm package using tsdx

### DISCUSSION THREAD

**Discussion by naptestdev** [vercel/next.js#35369] [Help]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

<img width="839" alt="image" src="https://user-images.githubusercontent.com/88878684/158524370-1e4977aa-5f9a-4395-b913-cb17b0defc2b.png">

### Describe the Bug

I use [tsdx](https://tsdx.io/) to create a react component library, but when I use any react hook, I get this same error.

<img width="726" alt="image" src="https://user-images.githubusercontent.com/88878684/158524447-e57ca3f9-05aa-40a8-958d-ced71732ae93.png">

### Expected Behavior

I want it to work normally if I use npm link to develop my package locally.

### To Reproduce

Maybe try to fix the nextjs compiler? When I use npm link with some other tool like vite or cra, it just works normally.

**Comment by icyJoseph** [comment]
Please refer to: https://github.com/facebook/react/issues/13991, tons of people report and fix their particular issues there.

This is most likely an issue with `tsdx` having its own react version installed, even if at the same version. I haven't used `tsdx` for a while now, but iirc you have to tell `tsdx` to use react from the top level module, or to not bundle it. 

To solve this issue you need to debug/read quite a bit, good luck!
